pkg/controller/certificates/trigger/policies/gatherer.go

```diff
@@ -46,30 +70,40 @@ func (g *Gatherer) DataForCertificate(ctx context.Context, crt *cmapi.Certificat
 		return Input{}, err
 	}
 
-	// Attempt to fetch the CertificateRequest resource for the current 'status.revision'.
+	// There can't be any available "current" certificate request if the
+	// certificate's revision has not been set yet. That is due to the fact
+	// that the certificate's revision field stays nil until the first
+	// certificate request (revision "1") has become ready.
+	if crt.Status.Revision == nil {
+		return Input{
+			Certificate: crt,
+			Secret:      secret,
+		}, nil
+	}
```

Comment:
I moved the `if` func in order to keep the [happy path unindented](https://maelvls.dev/go-happy-line-of-sight)